src/QsCompiler/CompilationManager/PerformanceTracking.cs

```diff
@@ -190,8 +190,17 @@ private static void InvokeTaskEvent(CompilationTaskEventType eventType, Task tas
 
             try
             {
-                var parent = GetTaskParent(task);
-                CompilationTaskEvent?.Invoke(eventType, parent?.ToString(), task.ToString());
+                if (details is null)
+                {
+                    var parent = GetTaskParent(task);
+                    CompilationTaskEvent?.Invoke(eventType, parent?.ToString(), task.ToString());
+                }
+                else
+                {
+                    var parent = task.ToString();
+                    CompilationTaskEvent?.Invoke(eventType, parent?.ToString(), details.ToString());
+                }
+
```

Comment:
```suggestion
                var parent = GetTaskParent(task);
                var taskId =
                    task.ToString() +
                    (details is null ? string.Empty : $"-{details}");

                CompilationTaskEvent?.Invoke(eventType, parent?.ToString(), taskId);
```

Reply:
This way we can keep the hierarchy protections and with the suggestion in my other comment it would make every rewrite step look like the following:
```
"OverallCompilation.RewriteSteps.SingleRewriteStep-FunctorGenerator"
"OverallCompilation.RewriteSteps.SingleRewriteStep-Monomorphization"
```

Reply:
Currently the rewrite steps look like this in the perf output:
```
"OverallCompilation.RewriteSteps": 427,
"OverallCompilation.RewriteSteps.Classically Controlled": 375,
"OverallCompilation.RewriteSteps.Conjugation Inlining": 2,
"OverallCompilation.RewriteSteps.Functor Generation": 13,
"OverallCompilation.RewriteSteps.Monomorphization": 35,
```

You're wanting it changed to:
```
"OverallCompilation.RewriteSteps": 427,
"OverallCompilation.RewriteSteps.SingleRewriteStep-Classically Controlled": 375,
"OverallCompilation.RewriteSteps.SingleRewriteStep-Conjugation Inlining": 2,
"OverallCompilation.RewriteSteps.SingleRewriteStep-Functor Generation": 13,
"OverallCompilation.RewriteSteps.SingleRewriteStep-Monomorphization": 35,
```
What is the advantage to having another enum value, SingleRewriteStep, and prefacing the rewrite step output with "SingleRewriteStep-"?

Reply:
Discussed offline: Using an explicit enum value better supports the task hierarchy, as well as future work on this feature.